### PR TITLE
Fareye superset Google auth 

### PR DESCRIPTION
**Pull request by mborkhat**
Google Auth and Superset Dashborad fix

Added google as oauth provider.
Added dashboard_id check to make it workable on chart pages also

### CODE REVIEW

File: superset/custom_security_manager.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# from flask import redirect, g, flash, request
```
Comment (saurabnigam) [inline superset/custom_security_manager.py:1]: Remove this file if not using

File: superset/config.py [hunk 2]
```diff
@@ -280,6 +283,23 @@ def _try_json_readsha(  # pylint: disable=unused-argument
 #    { 'name': 'Yahoo', 'url': 'https://open.login.yahoo.com/' },
 #    { 'name': 'Flickr', 'url': 'https://www.flickr.com/<username>' },
 
+# OAUTH_PROVIDERS = [
```
Comment (saurabnigam) [inline superset/config.py:286]: Remove this code if not using

File: superset-frontend/src/SqlLab/actions/sqlLab.js [hunk 3]
```diff
@@ -333,7 +333,7 @@ export function runQuery(query) {
     };
 
     return SupersetClient.post({
-      endpoint: `/superset/sql_json/${window.location.search}`,
+      endpoint: `/superset/sql_json/`,
```
Comment (saurabnigam) [inline superset-frontend/src/SqlLab/actions/sqlLab.js:336]: Have you verified this change?

**Review by saurabnigam — CHANGES_REQUESTED** [review]

**Comment by mborkhat** [comment]
Introducing Google Authentication in Superset.

Adding Dashboard check

File: superset-frontend/src/SqlLab/actions/sqlLab.js (hunk 3 above)
Comment (mborkhat) [reply superset-frontend/src/SqlLab/actions/sqlLab.js:336]: Yes

File: superset/config.py (hunk 2 above)
Comment (mborkhat) [reply superset/config.py:286]: Removed

File: superset/custom_security_manager.py (hunk 1 above)
Comment (mborkhat) [reply superset/custom_security_manager.py:1]: Removed

**Pull request merged**